src/modules/Elsa.Workflows.Core/Serialization/Converters/ActivityNodeConverter.cs

```diff
@@ -0,0 +1,48 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Elsa.Extensions;
+using Elsa.Workflows.Models;
+using Elsa.Workflows.Serialization.Helpers;
+
+namespace Elsa.Workflows.Serialization.Converters;
+
+/// <summary>
+/// Serializes the <see cref="ActivityNode"/> type and its descendant nodes based on the specified depth.
+/// </summary>
+/// <param name="depth">The number of levels of descendants to include. Defaults to 1.</param>
+public class ActivityNodeConverter(ActivityWriter activityWriter, int depth = 1, int level = 0) : JsonConverter<ActivityNode>
+{
+    /// <inheritdoc />
+    public override void Write(Utf8JsonWriter writer, ActivityNode value, JsonSerializerOptions options)
+    {
+        writer.WriteStartObject();
+        writer.WriteString("nodeId", value.NodeId);
+        writer.WriteString("port", value.Port);
+        writer.WritePropertyName("activity");
+        activityWriter.WriteActivity(writer, value.Activity, options, excludeChildren: true);
+
+        if (level < depth)
+        {
+            writer.WritePropertyName("children");
+            writer.WriteStartArray();
+
+            var nodeConverter = new ActivityNodeConverter(activityWriter, depth, 1);
```

Comment:
That's a great catch - in fact, it should have been `level + 1`.
After having updated that, I realized we could simplify a few things further, including the removal of re-stritching on the client side.

Originally, I wanted to render a graph of nodes to potentially support lazy loading of any children of any activity. However, this would require some significant changes on the client, so I implemented lazy loading of just the workflow definitions. Which removes the layered loading of levels completely, including the need for stitching. Instead, we can just rely on the WorkflowDefinitionActivity's activity provider to configure the serialization process to exclude its Root property; which means that on the client side, we have to load in just that root activity using the Graph API endpoint, which can return the desired structure as-is (no sticthing needed).